datafusion/src/physical_plan/expressions/sum.rs

```diff
@@ -153,9 +163,34 @@ macro_rules! typed_sum_delta_batch {
     }};
 }
 
+// TODO implement this in arrow-rs with simd
+// https://github.com/apache/arrow-rs/issues/1010
+fn sum_decimal_batch(
+    values: &ArrayRef,
+    precision: &usize,
+    scale: &usize,
+) -> Result<ScalarValue> {
+    let array = values.as_any().downcast_ref::<DecimalArray>().unwrap();
+
+    if array.null_count() == array.len() {
+        return Ok(ScalarValue::Decimal128(None, *precision, *scale));
+    }
+
+    let mut result = 0_i128;
+    for i in 0..array.len() {
+        if array.is_valid(i) {
+            result += array.value(i);
```

Comment:
I wonder if we need to check for overflow here as well?

Reply:
I just can check the overflow in the `eval` function.
When the result reaches the MAX of range (decimal(p,s)), but other values will affect the result.
If the MAX is `100` and now the result is 200 and the remaining values is [-100,-50], we can't throw overflow error or check overflow in the middle of calculation.